core/sql/common/CharType.cpp

```diff
@@ -1217,7 +1217,7 @@ void CharType::minMaxRepresentableValue(void* bufPtr,
   if (stringLiteral)
     {
       NABoolean isNull = FALSE;
-      NABoolean res = createSQLLiteral((const char *) bufPtr, *stringLiteral, isNull, h);
+      NABoolean res = createSQLLiteral((const char *) bufPtr  - getSQLnullHdrSize(), *stringLiteral, isNull, h);
```

Comment:
I still think move back the pointer will be the best approach here. I will add a comment to explicitly say the starting pos of the input bufPtr point to the position after the null header. And since the createSQLLiteral need the input buf starting address to be before null header, we need to move back. Otherwise, we have too much places to modify. What do you think?